src/classes/Bullet.ts

```diff
@@ -15,82 +16,83 @@ export class Bullet {
   public power: number;
   public destroyed: boolean;
 
-
   /**
-   * 
-   * @param {Tank} owner - owner of this Bullet. 
+   * @param {Tank} owner - owner of this Bullet.
    * @param {Number} id - unique id of the tank.
    * @param {Number} power - power of bullet.
-   * 
    */
-  constructor(owner: Tank, id: number, power?: number) {
+
+  public constructor(owner: Tank, id: number) {
     this.id = id;
     this.owner = owner;
     this.direction = owner.direction;
-    while(this.direction > 360) this.direction -= 360;
-    while(this.direction < 0) this.direction += 360;
+    while (this.direction > 360) {
+      this.direction -= 360;
+    }
```

Comment:
this.direction = this.direction % 360

Reply:
replace 360 with const

Reply:
originally planned to check the direction. I will delete this code. It is not necessary.